src/test/java/com/rst/bootcamp/InstructorsIntegrationTests.java

```
@@ -0,0 +1,142 @@
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.rst.bootcamp.model.Instructor;
+import com.rst.bootcamp.repository.InstructorRepository;
+
+@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
+@AutoConfigureMockMvc
+public class InstructorsIntegrationTests {
+
+	@Autowired
+	private MockMvc mockMvc;
+
+	@Autowired
+	InstructorRepository instructorRepo;
+
+	public static final MediaType APPLICATION_JSON_UTF8 = new MediaType(MediaType.APPLICATION_JSON.getType(),
+			MediaType.APPLICATION_JSON.getSubtype(), Charset.forName("utf8"));
+
+	@Test
+	public void test_addInstructor() throws Exception {
+		Instructor instructor = new Instructor();
+		instructor.setAdminName("A1");
+		instructor.setInstructorName("I1");
+
+		mockMvc.perform(post("/instructor").contentType(APPLICATION_JSON_UTF8).content(asJsonString(instructor)))
+				.andExpect(status().isCreated());
+	}
+
+	
+	public static String asJsonString(final Object obj) {
+		try {
+			return new ObjectMapper().writeValueAsString(obj);
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	@Test
+	public void test_addSubordinate() throws Exception {
+		try {
+			Instructor instructor = new Instructor();
+			instructor.setAdminName("A1");
+			instructor.setInstructorName("I2");
+
+			mockMvc.perform(post("/instructor").contentType(APPLICATION_JSON_UTF8).content(asJsonString(instructor)))
+					.andExpect(status().isCreated());
+
+			List<Instructor> instructors = instructorRepo.findAll();
+
+			mockMvc.perform(
+					post("/instructor/" + instructors.get(0).getId() + "/subordinate/" + instructors.get(1).getId())
+							.contentType(APPLICATION_JSON_UTF8))
+					.andExpect(status().isCreated());
+
+		} catch (Throwable e) {
```

empty catch block
do something.

---

Understood